modules/Kconfig.mcuboot

```diff
@@ -4,3 +4,15 @@ config SIGN_IMAGES
 	help
 	  Sign images for MCUBoot as integrated part of the build stages using
 	  the private key.
+
+config BOOT_SIGNATURE_KEY_FILE
+	string "MCUBoot PEM key file"
+	default "root-ec-p256.pem"
```

Comment:
Zephyr already have `MCUBOOT_SIGNATURE_KEY_FILE` for this purpose, so why can't that config setting not be used in this use-case ?
We already support signing with the keys specified in `MCUBOOT_SIGNATURE_KEY_FILE` config setting.

https://github.com/nrfconnect/sdk-zephyr/blob/7598e62f2e1a64191281f137ec553ed3017f1100/Kconfig.zephyr#L412-L436
```
config MCUBOOT_SIGNATURE_KEY_FILE
	string "Path to the mcuboot signing key file"
	....
	  This option should contain an absolute path to the same file
	  as the BOOT_SIGNATURE_KEY_FILE option in your MCUboot
	  .config. 
```

Today `MCUBOOT_SIGNATURE_KEY_FILE` takes precedence over `BOOT_SIGNATURE_KEY_FILE`:
https://github.com/nrfconnect/sdk-nrf/blob/3708c971b0a0b1ccec0792f35b70ebd01fd32fbd/modules/mcuboot/CMakeLists.txt#L128-L131
cause `BOOT_SIGNATURE_KEY_FILE` if `MCUBOOT_SIGNATURE_KEY_FILE` is not defined.

So if set, `MCUBOOT_SIGNATURE_KEY_FILE` will be used for signing.

Does this boil down to a documentation problem ?

Also, how do you know `root-ec-p256.pem` is a good default, if you don't know which signature type was specified when build MCUboot ?
```
config BOOT_SIGNATURE_TYPE_ECDSA_P256
config BOOT_SIGNATURE_TYPE_ED25519
config BOOT_SIGNATURE_TYPE_RSA
```
so maybe the safest is to let user specify what key file to use for signing, and bail out if he forgot to provide enough information to us.

Reply:
I can try to come up with a solution for using `MCUBOOT_SIGNATURE_KEY_FILE` directly. In its current state, things are really confusing, so the only option I see is to either remove `MCUBOOT_SIGNATURE_KEY_FILE` in for NCS, or to use it directly, so I will check out the last option.

We use ec-p256 as default since this is the default for MCUboot in NCS. That way, if a user has not changed the default configuration, the correct key will be used. IMO this improved the user experience since this key is often used during development. Also IMO it is better to have a guess at the default, rather than always require a user to perform some manual action.

Reply:
discussed oob.
Looking forward to see update